apps/omg_watcher_info/lib/omg_watcher_info/api/transaction.ex

```diff
@@ -77,55 +90,45 @@ defmodule OMG.WatcherInfo.API.Transaction do
   Given order finds spender's inputs sufficient to perform a payment.
   If also provided with receiver's address, creates and encodes a transaction.
   """
-  @spec create(UtxoSelection.order_t()) :: UtxoSelection.advice_t()
+  @spec create(TransactionCreator.order_t()) :: create_t()
   def create(order) do
-    utxos = DB.TxOutput.get_sorted_grouped_utxos(order.owner)
-    UtxoSelection.create_advice(utxos, order)
+    case order.owner
+         |> DB.TxOutput.get_sorted_grouped_utxos()
+         |> TransactionCreator.select_inputs(order) do
+      {:ok, inputs} ->
+        inputs
+        |> get_utxos_count()
+        |> create_transaction(inputs, order)
+
+      err ->
+        err
+    end
   end
 
-  @spec include_typed_data(UtxoSelection.advice_t()) :: UtxoSelection.advice_t()
-  def include_typed_data({:error, _} = err), do: err
-
-  def include_typed_data({:ok, %{transactions: txs} = advice}),
-    do: {
-      :ok,
-      %{advice | transactions: Enum.map(txs, fn tx -> Map.put_new(tx, :typed_data, add_type_specs(tx)) end)}
-    }
-
-  defp add_type_specs(%{inputs: inputs, outputs: outputs, metadata: metadata}) do
-    alias OMG.TypedDataHash
-
-    message =
-      [
-        create_inputs(inputs),
-        create_outputs(outputs),
-        [metadata: metadata || @empty_metadata]
-      ]
-      |> Enum.concat()
-      |> Map.new()
-
-    %{
-      domain: TypedDataHash.Config.domain_data_from_config(),
-      message: message
-    }
-    |> Map.merge(TypedDataHash.Types.eip712_types_specification())
+  defp get_utxos_count(inputs) do
+    inputs
+    |> Enum.map(fn {_, utxos} -> utxos end)
+    |> List.flatten()
+    |> Enum.count()
   end
 
-  defp create_inputs(inputs) do
-    empty_gen = fn -> %{blknum: 0, txindex: 0, oindex: 0} end
+  defp create_transaction(utxos_count, inputs, _order) when utxos_count > Transaction.Payment.max_inputs() do
+    transactions =
+      inputs
+      |> Enum.map(fn {_currency, inputs} ->
```

Comment:
Maybe use a different variable name here `inputs` is already the one you Map on.